### DISCUSSION TITLE
How to query related submissions for an asset?

### DISCUSSION THREAD

**Discussion by rungta** [verbb/formie#1764] [Q&A] [answered]
Hi @engram-design, we’re trying to do a cleanup job of all the duplicate assets created before this bug fix https://github.com/verbb/formie/issues/1729 was deployed. The idea was to run through all the assets on the volume, check if they are related to a submission, and if not, delete the asset. 

However, I’m unable to figure out the query to get this to work. Here’s an example:
- `19694` is the ID of an asset belonging to a submission.
- The asset belongs to a group field called `uploads`.
- `craft.formie.submissions.relatedTo(19694).exists()` returns `false`
- `craft.formie.submissions.relatedTo({ element: 19694, field: 'uploads' }).exists()` returns `false`

Any suggestions for the right query to figure out submissions related to an asset (both directly and via a group field)?

**Comment by engram-design** [comment, marked as answer]
Are your File Upload fields in a Group or Repeater field? If so, that'll be tricky as each block for those fields is an element on it's own - just like Matrix and Super Table. That means that the relationship is tied between the asset and the Group/Repeater Block element.

However, using `relatedTo` for non Group/Repeater fields (so File Uploads at the top level of the form) should be fine. But you're going to run into issues using the `field` reference, because these are Formie fields, and not Craft fields.

```twig
craft.formie.submissions.relatedTo({ element: 19694 }).exists()
```

Works for me on top-level fields. For nested fields, you can only really do that in PHP.

```php
\verbb\formie\elements\NestedFieldRow::find()->relatedTo(['element' => 19694])->exists();
```

But you'd want to be careful with that, as it's just matching against a "block", not a block and submission.

**Reply by rungta** [reply]
Thanks Josh, this worked. 🙏
